.github/workflows/php.yml

```diff
@@ -32,3 +34,59 @@ jobs:
 
     - name: Run test suite
       run: composer run-script test
+
+  analyse:
+
+    name: Static Analysis
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Validate composer.json and composer.lock
+      run: composer validate
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
+
+    - name: Install dependencies
+      if: steps.composer-cache.outputs.cache-hit != 'true'
+      run: composer install --prefer-dist --no-progress --no-suggest
+
+    - name: Run test suite
+      run: composer run-script analyse
```

Comment:
With composer scripts, you don't need the `run-script` part, you can run them with just `composer analyse`